src/main/java/org/opentest4j/FileInfo.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j;
+
+import java.io.Serializable;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+/**
+ * {@code FileInfo} is a pair of a file path and the contents of the file.
+ * <p>
+ * {@code FileInfo} is expected to be used as {@code actual} and {@code expected} values in {@code AssertionFailedError}
+ * <p>
+ * The semantics of the {@code path} can be defined by the users of this class.
+ *
+ * @author Marc Philipp
+ * @author Reinhold Degenfellner
+ * @since 1.2.1
+ */
+public class FileInfo implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+
+	private final String path;
+	private final byte[] contents;
+
+	/**
+	 * Constructs a {@code FileInfo} with a path to a file and the contents of the file.
+	 * <p>
+	 * Be cautious with large files (maybe limiting the size is a good idea)
+	 *
+	 * @param path the path to the file; must not be {@code null}
+	 * @param contents the contents of the file; must not be {@code null};
+	 */
+	public FileInfo(String path, String contents) {
+		if (contents == null) {
+			throw new NullPointerException("contents must not be null");
+		}
+		validatePath(path);
+		this.contents = contents.getBytes();
```

Comment:
isn't it better to not rely on default encoding ? what about using UTF8 ?

Reply:
I think that being able to choose the `CharSet` when retrieving the string with `public String getContentsAsString(Charset charset)`accounts for that.
If someone wants an explicit encoding they can use the `public FileInfo(String path, byte[] contents)`constructor.

Reply:
it won't work. 

because in order to reconstruct the string you must know the system charset otherwise you will get garbage.

also it is possible that the String (that can contain any Unicode characters) cannot be encoded correctly with the system charset.

ideally it would be better to keep the original String object or to enforce a will defined charset, because a byte[] doesn't carry the information about which charset was used

Reply:
this object is Serializable so it is possible that it will be deserialised someday on another jvm with different default charset

Reply:
`public String(byte[] bytes)`defaults to `Charset.defaultCharset()`.
Or you can use `getContentsAsString(CharSet.defaultCharset()`.
To my understanding the charset is always easy to get.

Reply:
if you are in a different jvm you won't be able to know it. 

Using getBytes() without an explicit charset is usually flagged as a code smell.

BTW, not a big deal. 
I just happened to this this patch.  It's up to the maintainers to decide. 

Thanks

Reply:
I agree with @eolivelli that we should not use the default charset when converting the `String` to a `byte[]`. I think the simplest option would be to remove this constructor and only keep the `byte[]` one. Thoughts?

Reply:
Removing the constructor sounds good to me. I wasn't sure anyway whether we would need it.
Thx for the input @eolivelli.